### ISSUE TITLE
Filter to extend the title block

### ISSUE THREAD

**Issue by roborourke** [WordPress/gutenberg#14009]
**Is your feature request related to a problem? Please describe.**

This is a similar request to https://github.com/WordPress/gutenberg/issues/5821 however without the need for back compat.

There are some use-cases in our projects at Human Made where we'd like to attach some extra inputs and information to the title block that aren't entirely suited to being blocks themselves, such as alternatives for A/B testing, short title for lists vs. long title for post pages and a more cohesive & visually representative "header" section including title, date, kicker, byline, social share buttons, comment count etc...

**Describe the solution you'd like**

There are a few options I could see:

1. Filter to insert components before & after the title block
2. Make title block a required part of a "header" area, blocks can be registered only for this section, and a template defined for it
3. Allow a block settings sidebar context for the title block

**Describe alternatives you've considered**

This could be handled via a custom sidebar plugin or metaboxes (and is being done that way in the meantime).

**Comment by draganescu** [comment]
I wonder if in the context of FSE this will still be needed or, instead of a filter for title, a feature of the post block would be better.

**Comment by karmatosed** [comment]
I tend to agree that this might be out dated now if just limits to the title block and maybe going beyond to the_content and also post block seems sensible. Let's get some thoughts on that with a view today.

**Comment by talldan** [comment]
I think it might still be quite hard to achieve something like this, but there are imperfect workarounds (e.g. adding tools in the Document Settings or Inspector).

There are an assortment of features like Block Bindings, Post Templates that should make something like this possible, but they're not quite evolved enough yet (as far as I'm aware).

For example, it is possible to build a post template that includes a heading block that represents a 'byline' underneath the Post Title. That byline heading could use bindings to connect to a `byline` post meta value.

From the post editor's normal view, that byline unfortunately wouldn't be editable or visible. It is possible to use the 'Show Template' option to make the byline visible, but my understanding is that it still wouldn't be editable. It'd nice to solve that, but then I'm also not sure 'Show Template' is the optimal editing view for posts.

So I think a nicer option would be a way to template the post editor's normal view (when 'Show Template' is turned off).

Just to add that blocks like 'byline' could be in the post content itself. That would probably work today, but I discount that as a good option as it limits layout possibilities.

For AB tests and Short Title, [the Block States issue might be the most relevant proposal I've seen](https://github.com/WordPress/gutenberg/issues/57719).
